doc/src/conf.py

```diff
@@ -0,0 +1,157 @@
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2018 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import os
+
+
+# -- General configuration ------------------------------------------------
+
+# minimal Sphinx version required.
+needs_sphinx = '1.5.3'
```

Comment:
Nice, I diddn't know about that one!

Reply:
It will certainly be useful to have this strict enforcement. Though it's a shame that, as far as I can tell at least, there isn't an equivalent blocker on versions that are too new, so we can hard-code our full version dependency range.

Reply:
The max version thing should disappear before too long, we can't go limping on with it for ever.